tests/functional/queries_with_sql_types.py

```diff
@@ -69,109 +75,99 @@ def test_character_types(create_drop_test_schema_fixture):
-            random_case_w = "".join(random.choice([k.upper(), k]) for k in w)
-            cur.execute('INSERT INTO schema_name.table_name VALUES(%s);' % random_case_w)
-            cur.execute('SELECT * FROM schema_name.table_name;')
-            r = cur.fetchmany(1)
-            assert r == [(outcome, )], "Failed for value: %s" % random_case_w
-            cur.execute('DELETE FROM schema_name.table_name;')
-
-    finally:
-        cur.execute('drop table schema_name.table_name;')
-        cur.execute('drop schema schema_name;')
-
-
-def test_numeric_constraint_violations(create_drop_test_schema_fixture):
+    cur = create_drop_test_schema_fixture
+
+    cur.execute(
+        'CREATE TABLE schema_name.table_name('
+        '   col boolean'
+        ');'
+    )
+
+    word_to_value = {
+        "TRUE": True,
+        "FALSE": False,
+        "'true'": True,
+        "'false'": False,
+        "'t'": True,
+        "'f'": False,
+        "'yes'": True,
+        "'no'": False,
+        "'y'": True,
+        "'n'": False,
+        "'on'": True,
+        "'off'": False,
+        "'1'": True,
+        "'0'": False,
+        "TRUE::boolean": True,
+        "FALSE::boolean": False,
+        "'yes'::boolean": True,
+        "'no'::boolean": False,
+    }
+
+    for (w, outcome) in word_to_value.items():
+        # it should work regardless of case so we randomly generate
+        # a string, which have both upper and lower case letters
+        random_case_w = "".join(random.choice([k.upper(), k]) for k in w)
+        cur.execute('INSERT INTO schema_name.table_name VALUES(%s);' % random_case_w)
+        cur.execute('SELECT * FROM schema_name.table_name;')
+        r = cur.fetchmany(1)
+        assert r == [(outcome, )], "Failed for value: %s" % random_case_w
+        cur.execute('DELETE FROM schema_name.table_name;')
+
+
+def test_numeric_constraint_violations(create_drop_test_schema_fixture: cursor):
     cur = create_drop_test_schema_fixture
     cur.execute('create table schema_name.table_name(si_col smallint, i_col integer, bi_col bigint);')
-    cur.execute('insert into schema_name.table_name values (%d, %d, %d);' % (-32768, -2147483648, -9223372036854775808))
-    cur.execute('insert into schema_name.table_name values (%d, %d, %d);' % (32767, 2147483647, 9223372036854775807))
+    args = [(-32768, -2147483648, -9223372036854775808),
+            (32767, 2147483647, 9223372036854775807)]
+    cur.executemany('insert into schema_name.table_name values (%s, %s, %s)', args)
 
-    try:
-        cur.execute('insert into schema_name.table_name values (%d, %d, %d);' % (32767, 2147483647, 9223372036854775809))
-    except pg.errors.NumericValueOutOfRange as e:
-        assert e.pgcode == '22003'
-        # assert e.pgerror == "ERROR: bigint out of range"
-    except pg.Error:
-        assert False
+    # await for NumericValueOutOfRange, will throw an error on different exception
+    with pytest.raises(NumericValueOutOfRange):
```

Comment:
is it possible to revert `assert e.pgcode == '22003'` part of code?

Reply:
done